app/uk/gov/hmrc/gform/validation/AddressValidation.scala

```diff
@@ -0,0 +1,99 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.validation
+import cats.Monoid
+import cats.implicits._
+import uk.gov.hmrc.gform.sharedmodel.form.FormDataRecalculated
+import uk.gov.hmrc.gform.sharedmodel.formtemplate._
+import uk.gov.hmrc.gform.validation.ComponentsValidator.{ errors, validateForbidden, validateRequired }
+import uk.gov.hmrc.gform.validation.ValidationUtil.ValidatedType
+import uk.gov.hmrc.gform.views.html.localisation
+import uk.gov.hmrc.gform.validation.ValidationServiceHelper.validationSuccess
+case class AddressValidation(data: FormDataRecalculated) {
+  import AddressValidation._
+  def validateAddress(fieldValue: FormComponent, address: Address)(data: FormDataRecalculated): ValidatedType[Unit] = {
+    val addressValueOf: String => Seq[String] = suffix => data.data.get(fieldValue.id.withSuffix(suffix)).toList.flatten
+
+    val validatedResult: List[ValidatedType[Unit]] = addressValueOf("uk") match {
+      case "true" :: Nil =>
+        List(
+          Monoid[ValidatedType[Unit]].combine(
+            validateRequiredField("street1", localisation("line 1"), fieldValue)(addressValueOf("street1")),
+            lengthValidation("street1", fieldValue)(addressValueOf("street1"))
+          ),
+          lengthValidation("street2", fieldValue)(addressValueOf("street2")),
+          lengthValidation("street3", fieldValue)(addressValueOf("street3")),
+          lengthValidation("street4", fieldValue)(addressValueOf("street4")),
+          validateRequiredField("postcode", localisation("postcode"), fieldValue)(addressValueOf("postcode")),
+          validateForbiddenField("country", fieldValue)(addressValueOf("country")),
+          postcodeLengthValidation("postcode", fieldValue)(addressValueOf("postcode"))
+        )
+      case _ =>
+        List(
+          Monoid[ValidatedType[Unit]].combine(
+            validateRequiredField("street1", localisation("line 1"), fieldValue)(addressValueOf("street1")),
+            lengthValidation("street1", fieldValue)(addressValueOf("street1"))
+          ),
+          lengthValidation("street2", fieldValue)(addressValueOf("street2")),
+          lengthValidation("street3", fieldValue)(addressValueOf("street3")),
+          lengthValidation("street4", fieldValue)(addressValueOf("street4")),
+          validateForbiddenField("postcode", fieldValue)(addressValueOf("postcode")),
+          validateRequiredField("country", localisation("Country"), fieldValue)(addressValueOf("country"))
+        )
+    }
+
+    Monoid[ValidatedType[Unit]].combineAll(validatedResult)
+  }
+}
+
+case object AddressValidation {
```

Comment:
`AddressValidation` should be `object` only.  

`case object` should be used only as part of `ADT` definition ([example here](https://github.com/hmrc/gform-frontend/blob/master/app/uk/gov/hmrc/gform/sharedmodel/formtemplate/AuthConfig.scala#L112-L118)), not stand alone as in this case. This stands for all standalone `case object`s in this PR.